source/room/MucManager.cpp

```diff
@@ -46,6 +49,35 @@ void MucManager::handleConnected()
     triggerNewMucSignal_ = true;
 }
 
+void MucManager::handleDataReceived(Swift::SafeByteArray data)
+{
+    std::string nodeData = Swift::safeByteArrayToString(data);
+    QString qData = QString::fromStdString(nodeData);
+
+    // Search from room name in disco#info
+
+    QString fromTag = XmlProcessor::getContentInTag("iq", "from", qData);
+
+    if(! fromTag.isEmpty())
+    {
+        QString query = XmlProcessor::getChildFromNode("query", qData);
+
+        if(! query.isEmpty())
+        {
+            QString catTag = XmlProcessor::getContentInTag("identity", "category", qData);
+            QString typeTag = XmlProcessor::getContentInTag("identity", "type", qData);
+
+            if ( catTag.contains("conference", Qt::CaseInsensitive) == true  &&
+                 typeTag.contains("text", Qt::CaseInsensitive) == true )
+            {
+                QString nameTag = XmlProcessor::getContentInTag("identity", "name", qData);
+
+                renameRoom(fromTag, nameTag);
+            }
+        }
+    }
+}
+
```

Comment:
I know, I also have done this in the existing code. But this does not work as expected every time :-(. Such constructs must be removed one by one. Sometimes, Swift calls the 'handleDataReceived' function with xml strings, cut in the middle randomly. This breaks the complete client implementation. Add a custom payload parser if you need data out of xml streams, which are not supported by swift out of the box:
https://swift.im/swiften/guide/

Reply:
I agree. I will look  to improve it.